test_scripts/API/Restructuring_OnResetTimeout/common_OnResetTimeout.lua

```diff
@@ -0,0 +1,559 @@
+
+c.jsonFileToTable = utils.jsonFileToTable
+c.tableToJsonFile = utils.tableToJsonFile
+c.cloneTable = utils.cloneTable
+c.modules = { "RADIO", "CLIMATE" }
+
+c.rpcs = {
+  "SendLocation",
+  "Alert",
+  "PerformInteraction",
+  "Slider",
+  "Speak",
+  "ScrollableMessage"
+}
+
+--[[ Common Functions ]]
+function c.getAppConfig(tbl)
+  if tbl then
+    local out = c.cloneTable(tbl.policy_table.app_policies.default)
+    out.moduleType = c.modules
+    out.groups = { "Base-4", "RemoteControl" }
+    out.AppHMIType = { "REMOTE_CONTROL" }
+    return out
+  else
+    return {
+      keep_context = false,
+      steal_focus = false,
+      priority = "NONE",
+      default_hmi = "NONE",
+      moduleType = c.modules,
+      groups = { "Base-4", "RemoteControl" },
+      AppHMIType = { "REMOTE_CONTROL" }
+    }
+  end
+end
+
+local function backupPreloadedPT()
+  local preloadedFile = commonFunctions:read_parameter_from_smart_device_link_ini("PreloadedPT")
+  commonPreconditions:BackupFile(preloadedFile)
+end
+
+local function updatePreloadedPT(pCountOfRCApps)
+  if not pCountOfRCApps then pCountOfRCApps = 2 end
+  local preloadedFile = commonPreconditions:GetPathToSDL()
+  .. commonFunctions:read_parameter_from_smart_device_link_ini("PreloadedPT")
+  local preloadedTable = c.jsonFileToTable(preloadedFile)
+  preloadedTable.policy_table.functional_groupings["DataConsent-2"].rpcs = json.null
+  preloadedTable.policy_table.functional_groupings["RemoteControl"].rpcs = {
+    SetInteriorVehicleData = {
+      hmi_levels = { "FULL", "BACKGROUND", "LIMITED", "NONE" }
+    },
+    SendLocation = {
+      hmi_levels = { "FULL", "BACKGROUND", "LIMITED", "NONE" }
+    },
+    DialNumber = {
+      hmi_levels = { "FULL", "BACKGROUND", "LIMITED", "NONE" }
+    },
+    DiagnosticMessage = {
+      hmi_levels = { "FULL", "BACKGROUND", "LIMITED", "NONE" }
+    },
+    ButtonPress = {
+      hmi_levels = { "FULL", "BACKGROUND", "LIMITED", "NONE" }
+    }
+  }
+  for i = 1, pCountOfRCApps do
+    local appId = config["application" .. i].registerAppInterfaceParams.fullAppID
+    preloadedTable.policy_table.app_policies[appId] = c.getAppConfig(preloadedTable)
+    preloadedTable.policy_table.app_policies[appId].AppHMIType = nil
+  end
+  c.tableToJsonFile(preloadedTable, preloadedFile)
+end
+
+function c.preconditions(isPreloadedUpdate, pCountOfRCApps)
+  if isPreloadedUpdate == nil then isPreloadedUpdate = true end
+  if isPreloadedUpdate == true then
+    backupPreloadedPT()
+    updatePreloadedPT(pCountOfRCApps)
+  end
+end
+
+local function restorePreloadedPT()
+  local preloadedFile = commonFunctions:read_parameter_from_smart_device_link_ini("PreloadedPT")
+  commonPreconditions:RestoreFile(preloadedFile)
+end
+
+--[[ @onResetTimeoutNotification: Notification from HMI  in case the results long processing on HMI
+--! @parameters:
+--! pRequestID - Id between HMI and SDL which SDL used to send the request for method in question, for which timeout needs to be reset.
```

Comment:
Enough description - `Request Id between HMI and SDL`  (which SDL used to send the request for method in question ?? do not understand what it means)